zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanData.java

```diff
@@ -160,36 +173,63 @@ private boolean isCustomPage(HttpMessage msg, CustomPage.Type cpType) {
     }
 
     /**
-     * Tells whether or not the message matches {@code CustomPage.Type.OK_200} definitions.
+     * Tells whether or not the message matches {@code CustomPage.Type.OK_200} definitions. Falls
+     * back to simply checking the response status code for "200 - Ok". Checks if the message
+     * matches {@code CustomPage.Type.ERROR_500} or {@code CusotmPage.Type.NOTFOUND_404} first, in
+     * case the user is trying to override something.
      *
      * @param msg the message that will be checked
      * @return {@code true} if the message matches, {@code false} otherwise
      * @since TODO Add version
      */
     public boolean isPage200(HttpMessage msg) {
-        return isCustomPage(msg, CustomPage.Type.OK_200);
+        if (isCustomPage(msg, CustomPage.Type.NOTFOUND_404)
+                || isCustomPage(msg, CustomPage.Type.ERROR_500)) {
+            return false;
```

Comment:
Not covered.

Reply:
I tried to get this morning but couldn't sort out the mocks/spy'ing and the fact that mocked classes can't be final.

Reply:
There shouldn't be any final class involved, one can cover these with e.g.: `given(context.isCustomPageWithFallback(msg, CustomPage.Type.NOTFOUND_404)).willReturn(true);`